__tests__/app.test.js

```diff
@@ -275,22 +277,604 @@ describe("app", () => {
+                  title: "Living in the shadow of a great man",
+                  topic: "mitch",
+                  author: "butter_bridge",
+                  body: "I find this existence challenging",
+                  created_at: expect.any(String),
+                  votes: 101,
+                  article_img_url: expect.any(String),
+                })
+              );
+            });
+        });
+    });
+
+    it("responds with an error code of 500 for all of other errors", () => {
+      jest.spyOn(db, "query").mockImplementation(() => {
+        throw new Error("Internal Server Error");
+      });
+
+      return Promise.all([
+        request(app).patch("/api/articles/1").send(testVote).expect(500),
+      ]).then(([res]) => {
+        expect(res.body.message).toEqual("Internal Server Error");
+      });
+    });
+  });
+
+  describe("GET /api/users endpoint", () => {
+    it("responds with a status 200 if successful", () => {
+      return request(app).get("/api/users").expect(200);
+    });
+    it("responds with an array of user objects", () => {
+      return request(app)
+        .get("/api/users")
+        .then((res) => {
+          let users = res.body.users;
+          expect(users).toBeInstanceOf(Array);
+        });
+    });
+    it("responds with an array of user objects with the correct length", () => {
+      return request(app)
+        .get("/api/users")
+        .then((res) => {
+          let users = res.body.users;
+          expect(users.length).toBe(4);
+        });
+    });
+    it("responds with an array of user objects with expected properties and values", () => {
+      return request(app)
+        .get("/api/users")
+        .then((res) => {
+          let users = res.body.users;
+          expect(users).toEqual(
+            expect.arrayContaining([
+              expect.objectContaining({
+                username: expect.any(String),
+                name: expect.any(String),
+                avatar_url: expect.any(String),
+              }),
+            ])
+          );
+        });
+    });
+    it("responds with a status 500 when an issue occurs", () => {
+      jest.spyOn(db, "query").mockImplementation(() => {
+        throw new Error("Internal Server Error");
+      });
+
+      return Promise.all([request(app).get("/api/users").expect(500)]).then(
+        ([res1]) => {
+          expect(res1.body.message).toEqual("Internal Server Error");
+        }
+      );
+    });
+  });
+
+  describe("GET /api/articles endpoint (now with queries!)", () => {
+    it("should be able to accept a 'topic' filter as a query", () => {
+      return request(app).get("/api/articles?topic=mitch").expect(200);
+    });
+    it("should only return matching topic articles when using topic filter", () => {
+      return request(app)
+        .get("/api/articles?topic=mitch")
+        .expect(200)
+        .then((res) => {
+          const articles = res.body.articles;
+          articles.forEach((article) => {
```

Comment:
Make sure your `.forEach( )` is protected with a length condition otherwise your tests will give false positive result for empty arrays.